drake/systems/plants/RigidBodySystem.h

```diff
@@ -130,7 +130,7 @@ class LoopConstraint : public LinearEqualityConstraint {
  * with additional sensors and actuators/forces
  * @concept{system_concept}
  */
-class DRAKERBSYSTEM_EXPORT RigidBodySystem {
+class DRAKERBSYSTEM_EXPORT RigidBodySystem : public std::enable_shared_from_this<RigidBodySystem> {
```

Comment:
I will review in a little bit, but do think we should try to avoid
enable_shared_from_this if at all possible.